src/gui/processing/models/qgsmodelcomponentgraphicitem.cpp

```diff
@@ -1219,13 +1219,18 @@ void QgsModelChildAlgorithmGraphicItem::activateAlgorithm()
     }
     else
     {
-      QMessageBox::warning( nullptr, QObject::tr( "Could not activate algorithm" ), QObject::tr( "The selected algorithm depends on other currently non-active algorithms.\n"
-                                                                                                 "Activate them them before trying to activate it.." ) );
+      int res = QMessageBox::warning( nullptr, QObject::tr( "Could not activate algorithm" ), QObject::tr( "The selected algorithm depends on other currently non-active algorithms.\n"
+                                                                                                 "These need to be activated before you can activate this algorithm.\n"
+                                                                                                 "Do you want to activate them also?" ),
+                                                                                                 QMessageBox::Yes | QMessageBox::No, QMessageBox::No);
+        if ( res != QMessageBox::Yes )
+          return;
+        // Reactivate also the algorithms the child algorithm depends on
+        model()->activateChildAlgorithm( child->childId(), true );     
```

Comment:
```suggestion
        model()->activateChildAlgorithm( child->childId(), true );
        emit requestModelRepaint();
```

But, as both if else conditions emit this signal, it could be moved after if else

Reply:
Thanks for the hint, Nico. So the function would be this, right?
```cpp
void QgsModelChildAlgorithmGraphicItem::activateAlgorithm()
{
  if ( const QgsProcessingModelChildAlgorithm *child = dynamic_cast<const QgsProcessingModelChildAlgorithm *>( component() ) )
  {
    if ( !model()->activateChildAlgorithm( child->childId() ) )
    {
      int res = QMessageBox::warning( nullptr, QObject::tr( "Could not activate algorithm" ), QObject::tr( "The selected algorithm depends on other currently non-active algorithms.\n"
                                                                                                 "These need to be activated before you can activate this algorithm.\n"
                                                                                                 "Do you want to activate them also?" ),
                                                                                                 QMessageBox::Yes | QMessageBox::No, QMessageBox::No);
      if ( res != QMessageBox::Yes )
        return;
      // Reactivate also the algorithms the child algorithm depends on
      model()->activateChildAlgorithm( child->childId(), true );
    }
    emit requestModelRepaint();
  }
}
```

Reply:
Looks good!